### PR TITLE
Quote plugin

### PR DESCRIPTION
**Pull request by JohnBee**
Quote people out of context for fun and for profit.
Simple quote plugin for Mathison. Enables simple adding and recalling of quotes. Quotes are stored and retrieved from a json file in plugin directory. Quote storage location could change and storage could be changed to mongoDB to be more scalable if usage increases.

### CODE REVIEW

**Comment by LordAro** [comment]
Well if it helps, those test failures aren't yours...

File: csbot/plugins/quote.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import json
+from ..plugin import Plugin
+
+class Quote(Plugin):
+    """Quote people out of context for fun and for profit.
+       Add quotes, recall quotes
+    """
+    class QUOTEERROR(Exception):
```
Comment (LordAro) [inline csbot/plugins/quote.py]: Should be CamelCase

File: csbot/plugins/quote.py [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        pass
+
+    def _quote(self,in_quote_num=None):
```
Comment (LordAro) [inline csbot/plugins/quote.py]: Various little PEP8 issues, like missing space between params

File: csbot/plugins/quote.py [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        quote = {}
+        if in_quote_num == None:
```
Comment (LordAro) [inline csbot/plugins/quote.py]: `if not in_quote_num:` ?

File: csbot/plugins/quote.py [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+        quote = {} #output
+
+        #open up quotes file
+        quotes = {}
+        try:
+            with open("quotes.json","r") as q_file:
+                quotes = json.loads(q_file.read())
+        except:
+            #quote file doesn't exist
+            quotes = {"quote_count": 1, '0': "Test quote please ignore."}
```
Comment (LordAro) [inline csbot/plugins/quote.py]: Should error here :)

File: csbot/plugins/quote.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+
+        #read the last key used, , this becomes the quote id, add one to it
+        num_of_quotes = 0
+        try:
+            num_of_quotes = quotes["quote_count"]
+        except:
```
Comment (LordAro) [inline csbot/plugins/quote.py]: `if "quote_count" in quotes:`

File: csbot/plugins/quote.py [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+
+    def _quote(self,in_quote_num=None):
+        quote = {}
+        if in_quote_num == None:
+            raise self.QUOTEERROR("No quote number passed.")
+        #check format of quote_numb
+        quote["ID"] = 0
+        try:
+            quote["ID"] = int(in_quote_num)
+        except:
```
Comment (LordAro) [inline csbot/plugins/quote.py]: Don't use these empty `except` catchalls - limit them to the exception type you want to catch

File: csbot/plugins/quote.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+            raise self.QUOTEERROR("Quote number passed not integer.")
+
+        #open the quotes file, and read the quote
+        quotes = {}
+        try:
+            with open("quotes.json","r") as q_file:
```
Comment (LordAro) [inline csbot/plugins/quote.py]: This isn't going to work at all, I'm afraid. Mathison is deployed on heroku so it has very limited amount of "state" and no files are allowed. Instead, you'll want to use the MongoDB integration - see cron and whois for usages

**Comment by JohnBee** [comment]
Fixed issues, moved to MongoDB

**Pull request closed without merging**